app/views/ladder/LadderView.coffee

```diff
@@ -74,16 +72,44 @@ module.exports = class LadderView extends RootView
     @loadLeague()
     @urls = require('core/urls')
 
-  checkForTournamentEnd: =>
+  setupMandateCheck: =>
     return if @destroyed
-    return false if me.isAdmin()
     $.get '/db/mandate', (data) =>
       return if @destroyed
-      if data?[0]?.currentTournament isnt 'magic-rush'
-        @tournamentEnd = true
-        @displayTabContent = 'display: none'
+      if @isTournamentBlocked data?[0], @courseInstance?.id, @level.get('slug')
+        unless me.isAdmin()
+          @tournamentEnd = true
+          @render()
       else
-        setTimeout @checkForTournamentEnd, 60 * 1000
+        setTimeout @setupMandateCheck, 60 * 1000
+
+  isTournamentBlocked: (mandate, courseInstanceID, levelSlug) =>
+    return unless mandate
+    tournament = _.find mandate?.currentTournament or [], (t) =>
+      t.courseInstanceID is courseInstanceID and t.level is levelSlug
+    if tournament
+      currentTime = @getCurrentDate() / 1000
+      if currentTime < tournament.startAt
+        delta = tournament.startAt - currentTime
+        console.log "Tournament start time: #{new Date(tournament.startAt * 1000)}, Time left: #{parseInt(delta / 60 / 60) }:#{parseInt(delta / 60) % 60}:#{parseInt(delta) % 60}"
+      else if currentTime > tournament.endAt
+        console.log "Tournament ended at: #{new Date(tournament.endAt * 1000)}"
+      return true unless tournament.startAt <= currentTime and tournament.endAt >= currentTime
+      delta = tournament.endAt - currentTime
+      console.log "Tournament end time: #{new Date(tournament.endAt * 1000)}, Time left: #{parseInt(delta / 60 / 60) }:#{parseInt(delta / 60) % 60}:#{parseInt(delta) % 60}"
+    else
+      return true if levelSlug in (mandate?.tournamentOnlyLevels or [])
+    return false
+
+  getServerDate: ->
+    res = $.ajax {async: false}
```

Comment:
I tested this line and got the warning:
`Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user’s experience. For more help http://xhr.spec.whatwg.org/`

I agree that it's worth checking the date via the server. Do you think there is a better way?

I'd recommend instead of calling a synchronous ajax every time, sending an ajax request on page load to the server and calculating a clock offset. Then use the `Date.now()` functionality corrected with the offset.

Reference: https://stackoverflow.com/a/1641147